tez-runtime-library/src/main/java/org/apache/tez/runtime/library/common/shuffle/impl/ShuffleManager.java

```diff
@@ -583,8 +583,15 @@ Fetcher constructFetcherForHost(InputHost inputHost, Configuration conf) {
       } else {
           alreadyCompleted = completedInputSet.get(input.getInputIdentifier());
       }
-      // Avoid adding attempts which have already completed or have been marked as OBSOLETE
-      if (alreadyCompleted || obsoletedInputs.contains(input)) {
+
+      // Avoid adding attempts which have already completed
+      if (alreadyCompleted) {
+        inputIter.remove();
+        continue;
+      }
+      // Avoid adding attempts which have been marked as OBSOLETE
+      if (isObsoleteInputAttemptIdentifier(input)) {
+        LOG.info("Skipping obsolete input: " + input);
```

Comment:
the original behavior didn't log when we skip an obsolete input, is this an intentional change?

I can see that we still don't log anything in ShuffleScheduler when the same happens, so this is more like a LOG.debug, or back to
```
if (alreadyCompleted || isObsoleteInputAttemptIdentifier(input)) {
```
which one do you prefer?

Reply:
I think it would be better to synchronize the behaviour of `ShuffleManager` and `ShuffleScheduler`. Modified to use a single if clause.